File: lab4/src/warehouse/equipments/Vehicle.java
package warehouse.equipments;

import enums.FuelType;
import enums.SupplyType;
import enums.VehicleType;
import request.SupplyRequestDetail;


public class Vehicle extends Equipment {
    VehicleType type;
    FuelType fuelType;

    public Vehicle(int id, String name, VehicleType type, FuelType fuelType, int count)
    {
        super(id, name, count);
        this.type = type;
        this.fuelType = fuelType;
    }

    @Override
    public Boolean isMatches(SupplyRequestDetail detail) {
        return detail.getSupplyType() == SupplyType.VEHICLE &&
               detail.getVehicleType() == type &&
               detail.getFuelType() == fuelType;
    }

    public  VehicleType getType() {
        return type;
    }

    public  FuelType getFuelType() {
        return fuelType;
    }
}

File: CourseProject/ILS_WPF/Models/Core/Equipments/Vehicle.cs
using ILS_WPF.Models.Core.Enums;
using ILS_WPF.Models.Core.Requests;

namespace ILS_WPF.Models.Core.Equipments
{
    public class Vehicle : Equipment
    {
        public VehicleType Type { get; set; }
        public FuelType FuelType { get; set; }

        protected Vehicle() { }

        public Vehicle(string name, VehicleType type, FuelType fuelType, int count)
        {
            Name = name;
            MaterialType = MaterialType.Vehicle;
            Type = type;
            FuelType = fuelType;
            Count = count;
        }

        public override bool IsMatches(SupplyRequestDetail detail)
            => detail.MaterialType == MaterialType.Vehicle &&
                   Type == detail.VehicleType && FuelType == detail.FuelType;
    }
}
